dynamic_programming/subset_sum.cpp

```diff
@@ -0,0 +1,194 @@
+/**
+ * @file
+ * @brief [Subset Sum Problem](https://en.wikipedia.org/wiki/Subset_sum_problem)
+ *                      (https://www.geeksforgeeks.org/subset-sum-problem-dp-25)
+ * @details
+ * Given a set of non-negative integers, and a value sum, determine if there is
+ * a subset of the given set with sum equal to given sum. 
+ *
+ * This problem can be solved using 2 methods:
+ * 1. Top down approach  : memoization approach
+ * 2. Bottom up appraoch : tabular approach
+ * Overall time complexity of coin change problem is O(n*sum)
+ *
+ * 
+ * Example 1:
+ * Input: set[] = {3, 34, 4, 12, 5, 2}, sum = 9
+ * Output: true
+ * Explanation: Return true because There is a subset (4, 5) with sum 9.
+ *
+ * Example 2:
+ * Input: set[] = {3, 34, 4, 12, 5, 2}, sum = 30
+ * Output: false
+ * Explanation: There is no subset that add up to 30.
+ *
+ * @author [Vinay Garg] (https://github.com/Vinay-Garg)
+ */
+
+#include <array>
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic programming algorithms
+ */
+namespace dynamic_programming {
+
+/**
+ * @namespace subset_sum_bottomup
+ * @brief bottom up implementation of subset sum problem
+ */
+namespace subset_sum_bottomup {
+/**
+ * @brief This is bottom up implemenation for checking if a subset 
+ * exists for the given sum value
+ * @tparam n 	: size of the set array
+ * @param set 	: an array of numbers in set of size<=n
+ * @param sum 	: subset sum value
+ * @returns true if there is a subset of set with sum equal to given sum
+ */
+template <size_t n>
+bool subset_sum(const std::array<int, n> &set, const int sum) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)).